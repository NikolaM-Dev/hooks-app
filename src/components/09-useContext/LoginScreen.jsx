import { useContext } from 'react';

import UserContext from './UserContext';

const LoginScreen = () => {
  const { setUser } = useContext(UserContext);

  const handleLogin = () =>
    setUser({
      id: 123,
      name: 'Nikola',
      email: 'nikola@gamil.com',
    });

  return (
    <>
      <h1>LoginScreen</h1>
      <hr />
      <button onClick={handleLogin} className="btn btn-primary">
        Login
      </button>
    </>
  );
};

export default LoginScreen;
