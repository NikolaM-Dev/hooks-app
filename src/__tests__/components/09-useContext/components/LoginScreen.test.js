import '@testing-library/jest-dom';
import { mount } from 'enzyme';

import LoginScreen from 'components/09-useContext/components/LoginScreen';
import UserContext from 'components/09-useContext/UserContext';

describe('Tests on <LoginScreen /> component', () => {
  const setUser = jest.fn();

  let wrapper = mount(
    <UserContext.Provider value={{ setUser }}>
      <LoginScreen />
    </UserContext.Provider>
  );

  beforeEach(() => {
    jest.clearAllMocks();
    wrapper = mount(
      <UserContext.Provider value={{ setUser }}>
        <LoginScreen />
      </UserContext.Provider>
    );
  });

  it('should show <LoginScreen /> correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should execute the setUser function with the expected argument', () => {
    wrapper.find('button').simulate('click');

    expect(setUser).toHaveBeenCalledWith({
      id: 123,
      name: 'Nikola',
      email: 'nikola@gamil.com',
    }); // User
    expect(setUser).toHaveBeenCalledTimes(1);
  });
});
