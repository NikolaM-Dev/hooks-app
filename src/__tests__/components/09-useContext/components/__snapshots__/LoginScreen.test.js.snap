// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on <LoginScreen /> component should show <LoginScreen /> correctly 1`] = `
Array [
  <h1>
    LoginScreen
  </h1>,
  <hr />,
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Login
  </button>,
]
`;
